services/web/server/src/simcore_service_webserver/workspaces/_workspaces_db.py

```diff
@@ -172,21 +181,21 @@ async def get_workspace_for_user(
     base_query = (
         select(
             *_SELECTION_ARGS,
-            access_rights_subquery.c.access_rights,
+            _access_rights_subquery.c.access_rights,
             my_access_rights_subquery.c.my_access_rights,
         )
         .select_from(
-            workspaces.join(access_rights_subquery).join(my_access_rights_subquery)
+            workspaces.join(_access_rights_subquery).join(my_access_rights_subquery)
         )
         .where(
             (workspaces.c.workspace_id == workspace_id)
             & (workspaces.c.product_name == product_name)
         )
     )
 
-    async with get_database_engine(app).acquire() as conn:
+    async with pass_or_acquire_connection(get_asyncpg_engine(app), connection) as conn:
         result = await conn.execute(base_query)
-        row = await result.first()
+        row = result.first()
```

Comment:
And this one. I would expect this one to be await maybe the `execute` above should be a `stream`?